app/src/main/java/com/doctor/yumyum/presentation/ui/mypage/MyPageFragment.kt

```diff
@@ -1,28 +1,78 @@
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
 
 class MyPageFragment : BaseFragment<FragmentMyPageBinding>(R.layout.fragment_my_page) {
 
-    private lateinit var viewModel: MyPageViewModel
+    private val viewModel by lazy {
+        ViewModelProvider(
+            this,
+            ViewModelProvider.NewInstanceFactory()
+        )[MyPageViewModel::class.java]
+    }
 
-    override fun onCreateView(
-        inflater: LayoutInflater, container: ViewGroup?,
-        savedInstanceState: Bundle?
-    ): View? {
-        return super.onCreateView(inflater, container, savedInstanceState)
+    private val gradeBadge: Map<String, Int> by lazy {
+        mapOf(
+            getString(R.string.grade_student) to R.drawable.ic_badge_student,
+            getString(R.string.grade_bachelor) to R.drawable.ic_badge_bachelor,
+            getString(R.string.grade_master) to R.drawable.ic_badge_master,
+            getString(R.string.grade_expert) to R.drawable.ic_badge_expert,
+            getString(R.string.grade_professor) to R.drawable.ic_badge_professor,
+        )
+    }
+    private val nextGrade: Map<String, String> by lazy {
+        mapOf(
+            getString(R.string.grade_student) to getString(R.string.grade_bachelor),
+            getString(R.string.grade_bachelor) to getString(R.string.grade_master),
+            getString(R.string.grade_master) to getString(R.string.grade_expert),
+            getString(R.string.grade_expert) to getString(R.string.grade_professor),
+            getString(R.string.grade_professor) to getString(R.string.grade_professor),
+        )
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        viewModel = ViewModelProvider(this).get(MyPageViewModel::class.java)
         binding.viewModel = viewModel
+        binding.lifecycleOwner = viewLifecycleOwner
+
+        CoroutineScope(Dispatchers.IO).launch {
+            viewModel.getUserInfo()
+        }
+
+        binding.apply {
+            myPageTvTasteSetting.setOnClickListener {
+                val tasteIntent = Intent(context, TasteActivity::class.java)
+                tasteIntent.putExtra(getString(R.string.taste_mode), true)
+                startActivity(tasteIntent)
+            }
+        }
+
+        viewModel.grade.observe(viewLifecycleOwner) {
+            gradeBadge[it]?.let { it1 -> binding.myPageIvBadge.setImageResource(it1) }
+            nextGrade[it]?.let { it1 -> binding.myPageTvNextLevel.text = it1 }
+        }
+
+        viewModel.point.observe(viewLifecycleOwner) { point ->
+            binding.myPageGraphOrange.layoutParams.apply {
+                val nextPoint: Int? = gradePoint[nextGrade[viewModel.grade.value]]
+                if (nextPoint != null) {
+                    width = binding.myPageGraphWhite.width * point / nextPoint
+                }
+            }
```

Comment:
좀 더 간결하게 표현할 수 있을 것 같습니당
```suggestion
            gradePoint[nextGrade[viewModel.grade.value]]?.let { nextPoint ->
                binding.myPageGraphOrange.layoutParams.width = binding.myPageGraphWhite.width * point / nextPoint
            }
```